src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiManifest.groovy

```diff
@@ -141,6 +146,44 @@ class JpiManifest extends Manifest {
     }
 
     private static listUpDependencies(Configuration c, boolean optional, StringBuilder buf) {
+        // The category attribute has been introduced in Gradle '5.3'
+        if (GradleVersion.current() >= GradleVersion.version('5.3')) {
+            listUpResolvedDependencies(c, optional, buf)
+        } else {
+            legacyListUpDependencies(c, optional, buf)
+        }
+    }
+
+    private static listUpResolvedDependencies(Configuration c, boolean optional, StringBuilder buf) {
+        def categoryAttribute = Attribute.of(Category.CATEGORY_ATTRIBUTE.name, String)
+
+        c.incoming.resolutionResult.root.dependencies.each { DependencyResult result ->
+            if (result.constraint || !(result instanceof ResolvedDependencyResult)) {
+                return
+            }
+            def selected = ((ResolvedDependencyResult) result).selected
+            // TODO: Find a better way to exclude platform dependencies
```

Comment:
I think this is ok to start with. A better solution is related to actually having a variant for JPI/HPI. Since this is not published yet (https://github.com/jenkinsci/gradle-jpi-plugin/pull/127#issuecomment-571107598) and will never be published for all plugins most likely, you could add a component metadata rule that adds the variant. But it's probably best to explore that together with the publishing of the variant.

Reply:
👍